src/main/java/uk/gov/hmcts/ccd/data/casedetails/query/CountCaseDetailsQueryBuilder.java

```diff
@@ -0,0 +1,25 @@
+package uk.gov.hmcts.ccd.data.casedetails.query;
```

Comment:
wonder whether it's better to have a buildCountQuery, buildSelectQuery, createCountQuery and createSelectQuery instead of introducing a hierarchy here

Reply:
That's what I tried initially, but I couldn't find a way to make it work without loosing the inner type of the query. Having a generic abstract class with subclasses to capture the type-specific code is the only way I found to preserve the `TypedQuery`. Also, it keeps the interface clean and keep the type specific functions concentrated in the factory.